gradio/queueing.py

```diff
@@ -345,81 +372,70 @@ async def clean_events(
                     if job.session_hash == session_hash or job._id == event_id:
                         job.alive = False
 
-        events_to_remove = []
-        for event in self.event_queue:
-            if event.session_hash == session_hash or event._id == event_id:
-                events_to_remove.append(event)
+        async with self.delete_lock:
+            events_to_remove: list[Event] = []
```

Comment:
I didn't want this to be running at the same time as `self.get_events`, because it may be that the first loop flags an event for removal, and then self.get_events pops the event off the queue into the active jobs, and then the second loop runs which actually removes the events from the queue, and it doesn't find the event and errors.

Could also do a "if event in queue: queue,remove(event)" check, if you think that's better.